### PR TITLE
[INTENG-4678]: Add support for Safari IAB safe-area

### PR DESCRIPTION
**Pull request by code-crusher**
**Description:**
As if now, in the Safari IAB, the banner gets overlapped by the bottom nav bar. This change ensure it's always above it.

**Reviewers:**
@rubinsingh

### CODE REVIEW

File: src/journeys_utils.js [hunk 1]
```diff
@@ -438,14 +439,94 @@ journeys_utils.animateBannerEntrance = function(banner) {
...
+		}
+	} else {
+		// bottom overscrolling is usually equivalent to half the banner size 
+		bannerIFrame.style.top = (windowHeight - bannerHeight) + (bannerHeight / 2) + "px";
+	}
+}
+
+// generic method to detect which devices should call journey_utls._dynamicallyRepositionBanner()
+journeys_utils._detectSafeAreaDevices = function() {
+	var isFormFactoriOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream;
```
Comment (rubinsingh) [inline src/journeys_utils.js]: @code-crusher can you please explain window.MSStream?

File: src/journeys_utils.js [hunk 2]
```diff
@@ -438,14 +439,94 @@ journeys_utils.animateBannerEntrance = function(banner) {
...
+		}
+	}
+}
+
+journeys_utils._dynamicallyRepositionBanner = function() {
+	journeys_utils.isSafeAreaEnabled = true;
+	// disable Journey animation to avoid lag when repositioning the banner 
+	document.getElementById('branch-banner-iframe').style.transition = "all 0s"
+	// make sure on the first journey load the position is correct
+	journeys_utils._resetJourneysBannerPosition(false, true);
```
Comment (rubinsingh) [inline src/journeys_utils.js]: @code-crusher on first load, why would the page already be scrolling?

Comment (rubinsingh) [reply src/journeys_utils.js]: Ah ignore, it's a signal to the function to check whether the page is already scrolling.

**Comment by rubinsingh** [comment]
@code-crusher looks great so far, I had a comment for your backend pr.

File: src/journeys_utils.js (hunk 1 above)
Comment (code-crusher) [reply src/journeys_utils.js]: Updated this to use `navigator.platform`

File: src/journeys_utils.js [hunk 3]
```diff
@@ -594,7 +652,8 @@ journeys_utils._getPageviewMetadata = function(options, additionalMetadata) {
 		"user_agent": navigator.userAgent,
 		"language": navigator.language,
 		"screen_width": screen.width || -1,
-		"screen_height": screen.height || -1
+		"screen_height": screen.height || -1,
+		"window_device_pixel_ratio": window.devicePixelRatio || 1
```
Comment (code-crusher) [inline src/journeys_utils.js]: Also passing the screen pixel ration to calculate the device screen size in the backend to calculate the logic for safe area flag.

**Comment by rubinsingh** [comment]
👍 but we need to make sure that circle-ci passes so let's hold off on the merge

**Comment by aaaronlopez** [comment]
👍

**Comment by rubinsingh** [comment]
@code-crusher feel free to merge when you feel it's appropriate. 

@JaveriaK helped to fix the circle-ci config.

**Comment by aaaronlopez** [comment]
@code-crusher we had to revert this PR before we release because the backend was out of sync. Can you open up a new PR with the changes you actually need here and leave it in PR state until we deploy? Then, QA can come it and check things out. cc @rubinsingh @GeneShay

**Comment by code-crusher** [comment]
Thanks for the heads up! I'm removing this Branch and creating a new one.

**Pull request merged**